qiskit_experiments/framework/composite/composite_experiment.py

```diff
@@ -40,10 +45,104 @@ def __init__(self, experiments, qubits, experiment_type=None):
+
+        Args:
+            backend: Target backend.
+            options: User provided runtime options.
+
+        Returns:
+            Transpiled circuit to execute.
+
+        Note:
+            This is transpile method for the composite experiment subclass.
+            This internally calls the transpile method of the nested experiments and
+            flattens the list of sub circuits generated by each experiment.
+            Note that transpile is called for individual circuit, and thus transpile
+            configurations and hook methods are separately applied.
+
+            No transpile configuration assumed for composite experiment object itself.
+        """
+        # Generate a set of transpiled circuits for each nested experiment and have them as a list.
+        # In each list element, a list of quantum circuit for nested experiment is stored.
+
+        # : List[List[QuantumCircuit]]
+        experiment_circuits_list = list(
+            map(lambda expr: expr.run_transpile(backend), self._experiments)
+        )
+
+        # This is not identical to the `num_qubits` when the backend is AerSimulator.
+        # In this case, usually a circuit qubit number is determined by the maximum qubit index.
+        n_qubits = 0
+        for circuits in experiment_circuits_list:
+            circuit_qubits = [circuit.num_qubits for circuit in circuits]
+            n_qubits = max(n_qubits, *circuit_qubits)
+
+        # merge circuits
+        return self._flatten_circuits(experiment_circuits_list, n_qubits)
+
+    def run_analysis(
+        self, experiment_data: ExperimentData, job: BaseJob = None, **options
+    ) -> ExperimentData:
+        """Run analysis and update ExperimentData with analysis result.
+
+        Args:
+            experiment_data: The experiment data to analyze.
+            job: The future object of experiment result which is currently running on the backend.
+            options: Additional analysis options. Any values set here will
+                override the value from :meth:`analysis_options` for the current run.
+
+        Returns:
+            An experiment data object containing the analysis results and figures.
+
+        Raises:
+            QiskitError: When the experiment data format is not for the composite experiment.
+
+        Note:
+            This is analysis method for the composite experiment subclass.
+            This internally calls the analysis method of the nested experiments and
+            outputs a representative data entry for the composite analysis.
+            Note that analysis is called for individual experiment data, and thus analysis
+            configurations and hook methods are separately applied.
+
+            No analysis configuration assumed for composite experiment object itself.
+        """
+        if not isinstance(experiment_data, self.__experiment_data__):
+            raise QiskitError("CompositeAnalysis must be run on CompositeExperimentData.")
```

Comment:
Now that you've copied this check to `CompositeAnalysis`, you can either remove it from here, or keep it here but in this case modify the error message, replace `CompositeExperimentData` with `self.__experiment_data__.__name__`, and similarly also for `CompositeAnalysis`.

Reply:
Thanks. You're right. The first one in composite experiment can be removed. Also message is updated. https://github.com/Qiskit/qiskit-experiments/pull/380/commits/d56612c2ec29f1c1e3385636e4a334599798e79b